universe/Condition.cpp

```diff
@@ -4957,13 +4959,13 @@ void DesignHasPartClass::Eval(const ScriptingContext& parent_context,
 }
 
 bool DesignHasPartClass::RootCandidateInvariant() const
-{ return (m_low->RootCandidateInvariant() && m_high->RootCandidateInvariant()); }
+{ return (m_low && m_high ? m_low->RootCandidateInvariant() && m_high->RootCandidateInvariant() : false); }
```

Comment:
need more complicated logic here to handle presence or absense of either independnetly. and if both aren't there, then it should return true.